### PR TITLE
MINOR: Change the input type of DescribeAclsResponse#aclsResources from Collection to Iterable to avoid creating useless collection

### PR DESCRIPTION
**Pull request by chris8099**
Modify DescribeAclsResponse, remove Collection and change to Iterable,
Modify KafkaApis, aclsResources method

### CODE REVIEW

**Comment by chris8099** [comment]
@chia7712  PTAL thanks

**Comment by chia7712** [comment]
@chris8099 Could you rebase code to include https://github.com/apache/kafka/commit/b9640a71c42ad117ed418209f87dc2962173469d?

**Comment by chris8099** [comment]
@chia7712 PTAL thanks~

**Comment by chia7712** [comment]
@chris8099 Could you revise the title? It seems to me the title is not clear.

**Comment by chris8099** [comment]
@chia7712 Okay, please check again.

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 1]
```diff
@@ -2509,12 +2509,10 @@ class KafkaApis(val requestChannel: RequestChannel,
             .setThrottleTimeMs(requestThrottleMs)))
       case Some(auth) =>
         val filter = describeAclsRequest.filter
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/KafkaApis.scala:2511]: Could we avoid this local variable?

**Review by chia7712 — COMMENTED** [review]
@chris8099 Thanks for your patch. one small question is left. Please take a look.

**Comment by chia7712** [comment]
@chris8099 Could you rebase code to include https://github.com/apache/kafka/commit/8db3b1a09af0bad274e07161336994610d616b35? Also, please fix the conflicting.

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 2]
```diff
@@ -2509,12 +2509,10 @@ class KafkaApis(val requestChannel: RequestChannel,
...
-        val returnedAcls = new util.HashSet[AclBinding]()
-        auth.acls(filter).forEach(returnedAcls.add)
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaApis.scala:2513]: Note that this means we no longer filter duplicates. We should make sure that doesn't cause issues.

Comment (chia7712) [reply core/src/main/scala/kafka/server/KafkaApis.scala:2513]: Good point! It seems to me the method ```aclsResources``` should de-duplicate as it tries to traverse all elements from input

**Comment by chia7712** [comment]
the related code is gone

**Pull request closed without merging**